windows/src/desktop/kmshell/main/Keyman.System.UpdateStateMachine.pas

```diff
@@ -753,18 +754,33 @@ procedure DownloadingState.Enter;
 var
   DownloadResult: Boolean;
   RetryCount: Integer;
+  FMutex: TKeymanMutex;
 begin
   // Enter DownloadingState
   bucStateContext.SetRegistryState(usDownloading);
 
   RetryCount := 0;
   DownloadResult := False;
+  FMutex := nil;
 
-  while (not DownloadResult) and (RetryCount < 3) do
-  begin
-    DownloadResult := DownloadUpdatesBackground;
-    if not DownloadResult then
-      Inc(RetryCount);
+  try
+    FMutex := TKeymanMutex.Create('KeymanDownloading');
```

Comment:
This should be outside the try-finally:

```suggestion
  FMutex := TKeymanMutex.Create('KeymanDownloading');
  try
```

Also, given we use the same name in multiple calls to the constructor, can we make that a constant at top of implementation section?

```
const
  KeymanDownloadMutexName = 'KeymanDownloading';
```